src/main/java/apoc/coll/Coll.java

```diff
@@ -127,6 +127,53 @@ public long indexOf(@Name("coll") List<Object> coll, @Name("value") Object value
     }
 
     @UserFunction
+    @Description("apoc.coll.set(coll, index, value) | set index to value")
+    public List<Object> set(@Name("coll") List<Object> coll, @Name("index") long index, @Name("value") Object value) {
+        if (coll == null) return null;
+        if (index < 0 || value == null) return coll;
+
+        List<Object> list = new ArrayList<>(coll);
+        list.set( (int) index, value );
+        return list;
+    }
+
+    @UserFunction
+    @Description("apoc.coll.insert(coll, index, value) | insert value at index")
+    public List<Object> insert(@Name("coll") List<Object> coll, @Name("index") long index, @Name("value") Object value) {
+        if (coll == null) return null;
+        if (index < 0 || value == null) return coll;
```

Comment:
value == null should be allowed

Reply:
@jexp I can remove this. Usually we see null in collections are bad but it's true that java does allow this so we can as well.